frontend/src/metabase-types/api/entity-id.ts

```diff
@@ -0,0 +1,24 @@
+const NANOID_ALPHABET =
+  "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const NANOID_LENGTH = 21;
+
+declare const __brand: unique symbol;
+type Brand<T, B> = T & { readonly [__brand]: B };
+type NanoID = Brand<string, "NanoID">;
+
+export const isNanoID = (id: unknown): id is NanoID => {
+  return (
+    typeof id === "string" &&
+    id.length === NANOID_LENGTH &&
+    new RegExp(`^[${NANOID_ALPHABET}]+$`).test(id)
+  );
+};
+
+export const asNanoID = (id: string): NanoID => {
```

Comment:
(note: looks like we don't reference `asNanoID` anywhere yet in this PR)

Reply:
with all of these new type errors, I think I'm about to. I'll make those changes first